examples/examples/fmt/yak_shave.rs

```diff
@@ -1,4 +1,6 @@
+#![allow(clippy::cognitive_complexity)]
```

Comment:
Personally, I'd prefer not to have clippy attributes in examples — the goal of the examples is to show the use of certain APIs, and I think adding stuff unrelated to the APIs we're trying to demonstrate just complicates them. I'd prefer making minor tweaks some of the examples to avoid the lints, if that's possible.

I can be persuaded otherwise, of course.

Reply:
Also, I _think_ the cognitive_complexity lint often fires on `tracing`'s macros, since it counts code inside of macro expansion against the number of lines in a function. If memory serves, this lint was removed from clippy's default lints for that reason. Can you make sure we still get this warning on the latest clippy version? Thanks!

Reply:
> Personally, I'd prefer not to have clippy attributes in examples — the goal of the examples is to show the use of certain APIs, and I think adding stuff unrelated to the APIs we're trying to demonstrate just complicates them. I'd prefer making minor tweaks some of the examples to avoid the lints, if that's possible.
> 
> I can be persuaded otherwise, of course.

I agree with this actually. In the case of the cognitive complexity lints I wasn't really sure how to make small changes which would accomplish this though, at least not without spending more time on it.

Reply:
> In the case of the cognitive complexity lints I wasn't really sure how to make small changes which would accomplish this though, at least not without spending more time on it.

Yeah, for that lint I _think_ there's nothing we can do about it, since it is triggered by basically all of `tracing`'s macros — see https://github.com/rust-lang/rust-clippy/issues/3900#issuecomment-526393180. I thought this lint was disabled by default in more recent versions of clippy. What version are you on? If it's the latest, this lint may have been turned back on inadvertently?

Reply:
@hawkw Just checked and I was using clippy `0.0.212`. Not sure why it was such an old version. I'll try updating to `0.0.302` and re-run against the code base.

Reply:
Actually I guess this is just the version of clippy installed via rustup. It should be `clippy-preview` but when I run `cargo clippy --version` it shows `0.0.212` so I don't know.